### DISCUSSION TITLE
Please help me remove Homebrew from my Mac.

### DISCUSSION THREAD

**Discussion by Ermiasmarino** [Homebrew/discussions#830] [Everyday usage]
I have Mojave on my mac and after running the uninstall script /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall.sh)", I got the following errors: 
```
Last login: Tue Feb 16 14:48:07 on console
Ermiass-MacBook-Pro:~ ermmy$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall.sh)"
Warning: This script will remove:
/Users/ermmy/Library/Caches/Homebrew/
/usr/local/Caskroom/
/usr/local/Cellar/
/usr/local/bin/brew -> /usr/local/bin/brew
Are you sure you want to uninstall Homebrew? This will remove your installed packages! [y/N] y
==> Removing Homebrew installation...
find: /usr/local/bin/.Tools-Service: Permission denied
Warning: Failed during: /usr/bin/find -E /usr/local/Frameworks /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var -regex .*/info/([^.][^/]*\.info|dir) -exec /bin/bash -c /usr/bin/install-info\ --delete\ --quiet\ {}\ "$(dirname\ {})/dir" ;
find: /usr/local/bin/.Tools-Service: Permission denied
Warning: Failed during: /usr/bin/find /usr/local/Frameworks /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var -type l -lname */Cellar/* -exec unlink {} ;
Warning: Failed to delete /usr/local/Caskroom
rm: /usr/local/Caskroom: Permission denied
Warning: Failed to delete /usr/local/Cellar
rm: /usr/local/Cellar: Permission denied
==> Removing empty directories...
Password:
==> /usr/bin/sudo /usr/bin/find /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var /usr/local/Caskroom /usr/local/Cellar /usr/local/Homebrew /usr/local/Frameworks -name .DS_Store -delete
==> /usr/bin/sudo /usr/bin/find /usr/local/bin /usr/local/etc /usr/local/include /usr/local/lib /usr/local/opt /usr/local/sbin /usr/local/share /usr/local/var /usr/local/Caskroom /usr/local/Cellar /usr/local/Homebrew /usr/local/Frameworks -depth -type d -empty -exec rmdir {} ;
Warning: Homebrew partially uninstalled (but there were steps that failed)!
To finish uninstalling rerun this script with `sudo`.
The following possible Homebrew files were not deleted:
/usr/local/.com.apple.installer.keep
/usr/local/Frameworks/
/usr/local/Homebrew/
/usr/local/bin/
/usr/local/etc/
/usr/local/include/
/usr/local/lib/
/usr/local/opt/
/usr/local/sbin/
/usr/local/share/
/usr/local/usr -> /usr/local/usr
/usr/local/var/
You may wish to remove them yourself.
```
How can I fix these problems and remove Homebrew cleanly so that I get the message that says I have successfully uninstalled Homebrew. Thanks.

**Comment by SMillerDev** [comment]
> Warning: Homebrew partially uninstalled (but there were steps that failed)!
To finish uninstalling rerun this script with sudo.

**Reply by Ermiasmarino** [reply]
I tried sudo /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall.sh)", but I got the following: 

/bin/bash: line 197: brew: command not found
Failed to locate Homebrew!

**Reply by SMillerDev** [reply]
If the install script does not work anymore you can take this advice:
```
The following possible Homebrew files were not deleted:
/usr/local/.com.apple.installer.keep
/usr/local/Frameworks/
/usr/local/Homebrew/
/usr/local/bin/
/usr/local/etc/
/usr/local/include/
/usr/local/lib/
/usr/local/opt/
/usr/local/sbin/
/usr/local/share/
/usr/local/usr -> /usr/local/usr
/usr/local/var/
You may wish to remove them yourself.
```

**Reply by Ermiasmarino** [reply]
I noticed there are files in usr/local/lib that are from other apps. Homebrew just messed up my computer. The developers of Homebrew should make Homebrew a self-contained app that you can remove completely with one click. This thing is all over the place. I didn't do anything wrong, but it has been showing me hundreds of errors. Just like it installed with one script, it must uninstall itself with one script. It is like someone who becomes too nice to enter your house, but refuses to leave your house once he entered by bringing all kinds of excuses. Shame on Homebrew developers.

**Reply by SMillerDev** [reply]
If you don't mess with permissions for the homebrew install directory that is exactly what the uninstall script does. The installer makes sure that all permissions are set up correctly, but if you change those the uninstaller won't be able to remove everything.

**Reply by Ermiasmarino** [reply]
I did not change anything. Take care.

**Reply by anmpog** [reply]
> If you don't mess with permissions for the homebrew install directory that is exactly what the uninstall script does. The installer makes sure that all permissions are set up correctly, but if you change those the uninstaller won't be able to remove everything.

While I understand that text does a poor job of conveying context, the replies you've offered in this thread are unhelpful and snippy.

**Comment by crossev** [comment]
To the original poster did you ever resolve this issue. I am having the same issue. I tried the uninstall script so that I could re-install a fresh homebrew and ended with the same error message. I tried to re-install homebrew to see if it would just overwrite, however now I am getting errors whenever I try to install anything with homebrew

**Comment by Pareech** [comment]
I was finally able to resolve the issue by following the instructions found in this comment https://github.com/Homebrew/discussions/discussions/3062#discussioncomment-2321335, I then followed the instructions to install Homebrew from here https://getgrav.org/blog/macos-monterey-apache-multiple-php-versions

**Reply by osalbahr** [reply]
Which part of the instructions in https://getgrav.org/blog/macos-monterey-apache-multiple-php-versions helped? It seems to basically instruct to use brew's install script anyways.
